modules/javafx.graphics/src/main/java/javafx/scene/text/TextLineInfo.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2024, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javafx.scene.text;
+
+import javafx.geometry.Rectangle2D;
+
+/**
+ * Provides the information about a text line in a text layout.
+ *
+ * @param start the start offset for the line
+ * @param end the end offset for the line (index of the last character + 1)
+ * @param bounds the bounds of the text line, in local coordinates:
+ * <ul>
+ *   <li>
```

Comment:
We usually indent with 4 spaces.

Reply:
in this economy, spaces are expensive ;-)

Reply:
what's a "tidy" and is it a part of our build/validation?

Reply:
The closing `</li>` tag is not required in HTML, and Javadoc doesn't seem to object. Is this a requirement of another tool?

Reply:
OK. I (mistakenly) thought that this was required. To answer Andy's question, [tidy](https://www.html-tidy.org/) is a toolto validate HTML5 for various formatting problems which can cause problems with some browsers. The JDK team used to periodically run it on their javadoc, but I don't think it is run continuously. Jon Gibbons filed [JDK-8184220](https://bugs.openjdk.org/browse/JDK-8184220) several years ago to "tidy" up our docs, which we did. This did not flag any missing `</li>` tags.

So: no change needed.